core/threshold/src/networking/tls.rs

```diff
@@ -36,63 +40,173 @@ pub struct ReleasePCRValues {
     pub pcr2: Vec<u8>,
 }
 
-/// TLS identity before the peer list is known
-#[derive(Debug, Clone)]
-pub struct BasicTLSConfig {
-    // public key certificate
-    pub cert: Pem,
-    // private key
-    pub key: Pem,
+#[derive(Debug)]
+pub struct AttestedTLSContext {
+    #[allow(dead_code)]
+    root_hint_subjects: Vec<DistinguishedName>,
+    verifiers: HashMap<String, (Arc<dyn ClientCertVerifier>, Arc<WebPkiServerVerifier>)>,
     // allowed software hashes
-    pub trusted_releases: Option<Arc<Vec<ReleasePCRValues>>>,
-    // are enclave images signed?
-    pub pcr8_expected: bool,
+    release_pcrs: Option<Arc<Vec<ReleasePCRValues>>>,
 }
 
-/// TLS identity after the peer list is known
-#[derive(Debug, Clone)]
-pub struct SendingServiceTLSConfig {
-    // public key certificate
-    pub cert: Pem,
-    // private key
-    pub key: Pem,
-    // A map from party subject names to party CAs. This map needs to be built
-    // by `build_ca_certs_map` to have the keys equal to subject names by
-    // construction.
-    pub ca_certs: HashMap<String, Pem>,
-    // allowed software hashes
-    pub trusted_releases: Option<Arc<Vec<ReleasePCRValues>>>,
-    // are enclave images signed?
-    pub pcr8_expected: bool,
+impl AttestedTLSContext {
+    pub fn new(
+        ca_certs: HashMap<String, Pem>,
+        release_pcrs: Option<Arc<Vec<ReleasePCRValues>>>,
+    ) -> anyhow::Result<Self> {
+        let verifiers = ca_certs
+            .iter()
+            .map(|(subject, ca_cert)| {
```

Comment:
Nitpick:L would it be possible to un-nest some of these closures? It is a bit hard to follow what is actually happening